### PR TITLE
feat: implement user research banner

### LINKED ISSUE #747: banner area for top menu and index nav
implements the user research banners
- in addition to the "gitHub icon slot" (`src/overrides/page-headers-side.js`)
 - shares a left border with it, user research banner below. 
 - user research banner is central for all sites. 

Invision sample (more screens available)
https://commercetools.invisionapp.com/console/share/HF1OY6M2Y3/483415275

### PR DESCRIPTION
**Pull request by TimonRey**
closes #747

### CODE REVIEW

File: packages/gatsby-theme-docs/src/components/user-research-banner.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  p {
+    font-size: ${designSystem.typography.fontSizes.small};
+  }
+`;
+
+const UserResearchBanner = () => {
+  return (
+    <Banner>
+      <div>
+        <img width="22px" height="22px" src={ctLogo}></img>
```
Comment (TimonRey) [inline packages/gatsby-theme-docs/src/components/user-research-banner.js:25]: Not sure where to get a svg or an image of the ct-logo without space around it 🤔

File: packages/gatsby-theme-docs/src/overrides/page-header-side.js [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+// eslint-disable-next-line react/display-name
+export default () => {
+  return (
+    <div>
+      <div
+        css={css`
+          padding-bottom: ${designSystem.dimensions.spacings.m};
+        `}
+      >
+        <svg viewBox="0 0 16 16" width="24" height="24">
```
Comment (TimonRey) [inline packages/gatsby-theme-docs/src/overrides/page-header-side.js:16]: I also implemented the Github icon so that you can see how the upper right corner behaves with multiple components rendered...but I guess we should remove that before merging.

File: .changeset/six-trains-switch.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@commercetools-docs/gatsby-theme-docs': patch
```
Comment (nkuehn) [inline .changeset/six-trains-switch.md:2]: A new but backwards compatible feature is a minor, not a patch. 
https://semver.org/

File: packages/gatsby-theme-docs/src/components/top-menu.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import { css } from '@emotion/core';
 import SpacingsInline from '@commercetools-uikit/spacings-inline';
 import SpacingsStack from '@commercetools-uikit/spacings-stack';
 import { designSystem } from '@commercetools-docs/ui-kit';
+import UserResearchBanner from './user-research-banner';
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/src/components/top-menu.js:9]: I remember that we wanted to make this a generic place to post banners ("top-menu-banner") and not hardwire the specific current content into the code structure. 

I suggest to make the banner overridable by placing an empty "top-menu-banner-area" into the overrides folder which then imports and uses the actual user-research-banner component.

File: packages/gatsby-theme-docs/src/components/user-research-banner.js (hunk 1 above)
Comment (nkuehn) [reply packages/gatsby-theme-docs/src/components/user-research-banner.js:25]: I think we should have the CT logos made available by the docs ui kit as components.  Concerning the spacing around it you have to ask Sven, I remember someone telling me that the spacing is required and sven's design does actually have more spacing as opposed to this implementation (at least to the top and bottom, the left spacing could have to be overridden in CSS.

File: packages/gatsby-theme-docs/src/components/user-research-banner.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        <p>We want your feedback</p>
+        <Link
+          href={'https://ok.commercetools.com/free-trial'}
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/src/components/user-research-banner.js:28]: This is the wrong URL.  Merchant Center and all the other existing places refer to https://ok.commercetools.com/user-research-program

File: packages/gatsby-theme-docs/src/layouts/internals/footer.js [hunk 6]
```diff
@@ -190,6 +190,9 @@ const LayoutFooter = () => {
                 {node.label}
               </GlobalNavigationLink>
             ))}
+            <GlobalNavigationLink href="https://ok.commercetools.com/free-trial">
```
Comment (nkuehn) [inline packages/gatsby-theme-docs/src/layouts/internals/footer.js:193]: wrong URL  here, too  -->  https://ok.commercetools.com/user-research-program

File: packages/gatsby-theme-docs/src/overrides/page-header-side.js (hunk 2 above)
Comment (nkuehn) [reply packages/gatsby-theme-docs/src/overrides/page-header-side.js:16]: The specific example here should not be directly in the theme but e.g. in the API Smoke Test site as an override so we can test and compare two sites with and without the override.   In the smoke test site it should actually stay (you can use a different Icon but it would stay as a regression test)

**Review by nkuehn — DISMISSED** [review]
See comments, plus:

Spacing and typography of the implementation does not match the design. 

Design:
image


Implementation: 
image

File: packages/gatsby-theme-docs/src/components/top-menu.js (hunk 4 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/components/top-menu.js]: Hmm, right. Sorry I had it different in mind 👍

File: .changeset/six-trains-switch.md (hunk 3 above)
Comment (TimonRey) [reply .changeset/six-trains-switch.md]: Thanks for the link 👍

File: packages/gatsby-theme-docs/src/components/user-research-banner.js (hunk 5 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/components/user-research-banner.js]: Right 😅

**Comment by emmenko** [comment]
Each website should override the `page-header-side.js`, that's the component to use (for example https://github.com/commercetools/merchant-center-application-kit/blob/master/website/src/%40commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js). The theme should not implement this feature.

Then there is the question where expose the banner component. We can expose it from the theme, or from the kit. The important thing is that the theme does not implement this.

**Comment by nkuehn** [comment]
(from today's daily for documentation)

> Each website should override the `page-header-side.js`,
No, each website should be able to override the side banner but the theme should provide the override "slot" pre-filled with the centrally defined one.  

The "GitHub icon" override area needs to be separate to allow the kit to roll out a different standard banner independently of whether a side has a gitHub icon or something like that.

File: packages/gatsby-theme-docs/src/components/user-research-banner.js (hunk 1 above)
Comment (emmenko) [reply packages/gatsby-theme-docs/src/components/user-research-banner.js:24]: You should use the one from appkit `@commercetools-frontend/assets` package, which is an SVG.

https://github.com/commercetools/merchant-center-application-kit/blob/636fc45c0d53544a90b0977b2212ec3a8aec8b94/packages/application-shell/src/components/app-bar/app-bar.tsx#L9

File: packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+// A React component to be rendered on the right side of the top-menu
+export default () => null;
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js:2]: According to Nikolaus, the banner should be rendered by default here.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js [hunk 8]
```diff
@@ -4,13 +4,17 @@ import { designSystem } from '@commercetools-docs/ui-kit';
...
+    ${designSystem.dimensions.spacings.xl} 0;
+  border-left: 1px solid ${designSystem.colors.light.borderPrimary};
 
   @media screen and (${designSystem.dimensions.viewports.largeTablet}) {
     display: block;
     width: ${designSystem.dimensions.widths.pageNavigation};
   }
   @media screen and (${designSystem.dimensions.viewports.laptop}) {
-    width: ${designSystem.dimensions.widths.pageNavigationSmall};
+    width: ${designSystem.dimensions.widths.pageHeaderSideSmall};
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:17]: Why the changes here? 🤔

File: packages/gatsby-theme-docs/src/components/user-research-banner.js (hunk 1 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/components/user-research-banner.js:24]: Ah, thanks!! 👍

File: packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js (hunk 7 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js:2]: So we don't need to override it in the websites at all? 🤔

Comment (emmenko) [reply packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js:2]: In the smoke tests websites you still need to override the other placeholder (with the github link) to have them visually tested.
The smoke tests websites do not need to override the banner, it's only useful to Nikolaus for the future.

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js (hunk 8 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:17]: Bildschirmfoto 2020-11-24 um 01 04 49

I noticed with this perticular screen size that the width of the layout is too wide for the banner...as you can see you can't see the end of the border or half of the external icon. So I thought it's better to change the layout width so that this can't occur with other components with full width that could be rendered in this layout.
What do you think? 🤔

File: packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js (hunk 7 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/overrides/top-menu-banner-area.js]: Ah right, sorry I ment that we don't need to override the `top-menu-banner-area.js` right?
a456034

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js (hunk 8 above)
Comment (nkuehn) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:17]: We're definitely not changing the Layout just for a Banner with limited lifetime. 

For me the Banner should behave like normal HTML layout, i.e. if a line is too short the text wraps.  If the business owners don't like that they need to come up with a differnt text.  

By the way: "user research program" is not a given name, so please don't capitalize it.  (or, if it's chose to be a given name, then capitalize it completely).

Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:17]: Agree, the layout must not change. The link text should just wrap.

File: packages/gatsby-theme-docs/src/components/user-research-banner.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    font-size: ${designSystem.typography.fontSizes.small};
+  }
+`;
+
+const UserResearchBanner = () => {
+  return (
+    <Banner>
+      <div
+        css={css`
+          padding: 8px 0 4px 0;
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/user-research-banner.js:20]: These values should not be hard coded.

File: packages/gatsby-theme-docs/src/components/user-research-banner.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        `}
+      >
+        <LogoSVG />
+      </div>
+      <p>We want your feedback</p>
+      <Link
+        href={'https://ok.commercetools.com/user-research-program'}
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/user-research-banner.js:27]:
```suggestion
        href="https://ok.commercetools.com/user-research-program"
```

File: packages/ui-kit/src/design-system.js [hunk 11]
```diff
@@ -136,6 +136,7 @@ export const dimensions = {
     pageContentWideWithMargins: `calc(${pageWidthSmall} + ${asideColumnWidth} + 32px * 3)`,
     pageContentWideWithMarginsMax: `calc(${pageWidth} + ${asideColumnWidth} + 32px * 3)`,
     pageContentWithMarginsAndPageNavigation: `calc(${pageWidth} + 32px * 2 + ${navbarWidth})`,
+    pageHeaderSideSmall: '175px',
```
Comment (emmenko) [inline packages/ui-kit/src/design-system.js:139]: This can be removed

File: websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    <div>
+      <div
+        css={css`
+          padding-bottom: ${designSystem.dimensions.spacings.m};
+        `}
+      >
+        <GitHubIcon />
+      </div>
+      <UserResearchBanner />
+    </div>
```
Comment (emmenko) [inline websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js:20]:
```suggestion
    <SpacingsStack size="m">
      <div>
        <GitHubIcon />
      </div>
      <UserResearchBanner />
    </SpacingsStack>
```

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js [hunk 13]
```diff
@@ -4,16 +4,29 @@ import { designSystem } from '@commercetools-docs/ui-kit';
 const LayoutPageHeaderSide = styled.aside`
   grid-area: page-header-side;
   display: none;
+  padding: 0 ${designSystem.dimensions.spacings.m};
+  margin: ${designSystem.dimensions.spacings.m} 0
+    ${designSystem.dimensions.spacings.xl} 0;
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:9]:
```suggestion
  margin: ${designSystem.dimensions.spacings.m} 0;
```

Comment (TimonRey) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js:9]: But the spacing for margin-bottom should be bigger than for the top 🤔

File: websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js (hunk 12 above)
Comment (TimonRey) [reply websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js]: Right, thanks 👍

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js (hunk 13 above)
Comment (emmenko) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js]: That spacing you're talking about (I suppose) is something else and depends on the page heading.
For example

image

The margin here should be consistent top/bottom, just for safety measures.

File: websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+
+// A React component to be rendered in the top-right corner of a content page
+// eslint-disable-next-line react/display-name
+export default () => {
+  return (
+    <SpacingsStack size="m">
+      <div>
+        <GitHubIcon />
+      </div>
+      <UserResearchBanner />
```
Comment (emmenko) [inline websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js:14]: Sorry for the back and forth.

The banner should be included in the docs theme by default, and should be rendered as a separate overrides here:

https://github.com/commercetools/commercetools-docs-kit/blob/b4063e6518eb4398b6a8f8e75ea7fb6739795d01/packages/gatsby-theme-docs/src/layouts/content.js#L56-L58

```js
<LayoutPageHeaderSide>
  <PlaceholderPageHeaderSide />
  <PlaceholderPageHeaderSideBannerArea />
</LayoutPageHeaderSide>
```

File: websites/docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js [hunk 15]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { UserResearchBanner } from '@commercetools-docs/gatsby-theme-docs';
+
+// A React component to be rendered in the top-right corner of a content page
+// eslint-disable-next-line react/display-name
+export default () => {
+  return <UserResearchBanner />;
+};
```
Comment (emmenko) [inline websites/docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js:8]: This override is not necessary.

**Review by emmenko — CHANGES_REQUESTED** [review]

File: packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js (hunk 13 above)
Comment (TimonRey) [reply packages/gatsby-theme-docs/src/layouts/internals/layout-page-header-side.js]: Allright 👍

File: websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js (hunk 14 above)
Comment (TimonRey) [reply websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js:14]: 🙈

File: packages/gatsby-theme-docs/index.js [hunk 16]
```diff
@@ -12,6 +12,7 @@ export {
   ThemeProvider,
   SideBySide,
   FullWidthContainer,
+  UserResearchBanner,
```
Comment (emmenko) [inline packages/gatsby-theme-docs/index.js:15]: We don't need to export it for now.

File: .changeset/six-trains-switch.md [hunk 17]
```diff
@@ -0,0 +1,6 @@
+---
+'@commercetools-docs/gatsby-theme-docs': minor
+'@commercetools-docs/ui-kit': patch
```
Comment (emmenko) [inline .changeset/six-trains-switch.md:3]:
```suggestion
'@commercetools-docs/gatsby-theme-docs': minor
```

File: .changeset/six-trains-switch.md [hunk 18]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+A user research banner was implemented for all sites
```
Comment (emmenko) [inline .changeset/six-trains-switch.md:6]:
```suggestion
Render a User Research banner in all content pages.
```

**Review by emmenko — COMMENTED** [review]
Thanks, looks good now! 👌

File: .changeset/calm-bulldogs-repair.md [hunk 19]
```diff
@@ -1,6 +1,5 @@
 ---
 '@commercetools-docs/gatsby-theme-docs': minor
-'@commercetools-docs/ui-kit': patch
 ---
 
-The RssFeed is now one single table which shows the newest release notes of all feeds
```
Comment (emmenko) [inline .changeset/calm-bulldogs-repair.md:6]: This is a different changeset, you should not change this.

Comment (TimonRey) [reply .changeset/calm-bulldogs-repair.md]: Sry for that, I accidently changed the wrong changeset...thanks 👍 
d038274

**Review by emmenko — APPROVED** [review]

**Comment by TimonRey** [comment]
From the design review: The margin between the last link and the banner in the top menu should be 24px
92080de

**Comment by emmenko** [comment]
You should update the spacing stack then: https://github.com/commercetools/commercetools-docs-kit/pull/756/files#diff-aa31a728488c0176ea0bb9d3f1979be93cad0c49f0c40cce90511d0328185d89R59

**Comment by TimonRey** [comment]
Ich changed the spacings now 👍 
9666e7f

File: packages/gatsby-theme-docs/src/components/top-menu.js [hunk 20]
```diff
@@ -201,12 +202,15 @@ const TopMenu = (props) => {
...
-                  ))}
+                <SpacingsStack scale="l">
+                  <SpacingsStack scale="s">
+                    {data.allTopSideMenuYaml.nodes.map((node) => (
+                      <GlobalNavigationLink href={node.href} key={node.id}>
+                        {node.label}
+                      </GlobalNavigationLink>
+                    ))}
+                  </SpacingsStack>
+                  <TopMenuBannerArea />
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/top-menu.js:213]: 💯

**Review by emmenko — APPROVED** [review]

**Comment by TimonRey** [comment]
New suggestions from @nkuehn and @Shecki:
02bfe98

File: packages/gatsby-theme-docs/src/components/user-research-banner.js [hunk 21]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import css from '@emotion/css';
```
Comment (emmenko) [inline packages/gatsby-theme-docs/src/components/user-research-banner.js:3]:
```suggestion
import { css } from '@emotion/react';
```

File: websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js [hunk 22]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
-import GitHubIcon from '../../../icons/github.svg';
+import SvgGithub from '../../../icons/generated/Github';
```
Comment (TimonRey) [inline websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js:2]: Don't know why I can't import it directly over `../icons` 🤔

Comment (emmenko) [reply websites/api-docs-smoke-test/src/@commercetools-docs/gatsby-theme-docs/overrides/page-header-side.js]: Fixed: https://github.com/commercetools/commercetools-docs-kit/pull/756/commits/c5975bbe8b5cd3a888af2de6d8f54d1d41a84fb8
